lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/LibertyVersionDownloadUtil.java

```diff
@@ -0,0 +1,274 @@
+     * Saves the XML content to the specified local cache file path.
+     *
+     * @param cacheFilePath path to save file
+     * @param xmlContent    The XML content to save.
+     */
+    private static void saveToLocalCache(String cacheFilePath, String xmlContent) {
+        Path filePath = Paths.get(cacheFilePath);
+        try {
+            // Ensure the parent directories exist before writing the file.
+            Files.createDirectories(filePath.getParent());
+            Files.writeString(filePath, xmlContent);
+            LOGGER.fine("Successfully saved content to local cache.");
+        } catch (IOException e) {
+            LOGGER.warning("Failed to save content to local cache: " + e.getMessage());
+        }
+    }
+
+    /**
+     * A helper method to parse the 'latest' version from a maven-metadata.xml string.
+     *
+     * @param xmlContent The XML content as a string.
+     * @return The version string, or null if not found or on parsing error.
+     */
+    private static String parseVersionFromXml(String xmlContent) {
+        try {
+            Document doc = DocumentUtil.getDocument(xmlContent);
+            doc.getDocumentElement().normalize();
+
+            // Find the <latest> tag.
+            NodeList nodeList = doc.getElementsByTagName("latest");
+            if (nodeList.getLength() > 0) {
+                return nodeList.item(0).getTextContent();
+            }
+        } catch (Exception e) {
+            LOGGER.warning("Failed to parse metadata XML: " + e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * get url from maven repo
+     * if locale is provided, use different url
+     * @return schemaURl parsed
+     */
+    private static String parseURL(String url, String urlWithLocale) {
+        String rtVersion = SettingsService.getInstance().getLatestRuntimeVersion();
+        String schemaURL;
+        if (SettingsService.getInstance().getCurrentLocale() == null || SettingsService.getInstance().getCurrentLocale().equals(Locale.US)
+                || SettingsService.getInstance().getCurrentLocale().equals(Locale.ENGLISH) || urlWithLocale == null) {
+            schemaURL = url.replace(VERSION, rtVersion);
+        } else {
+            Locale currentLocale = SettingsService.getInstance().getCurrentLocale();
+            // liberty published file names contain only language code as locale for all locales except Portuguese (Brazil) and Traditional Chinese
+            // filename samples
+            // https://repo1.maven.org/maven2/io/openliberty/features/open_liberty_schema_es/25.0.0.8/open_liberty_schema_es-25.0.0.8.xsd
+            // https://repo1.maven.org/maven2/io/openliberty/features/open_liberty_schema_pt_BR/25.0.0.8/open_liberty_schema_pt_BR-25.0.0.8.xsd
+            schemaURL = urlWithLocale.replace(VERSION, rtVersion).replace("$LOCALE", currentLocale.getLanguage());
+            if (Locale.TAIWAN.equals(currentLocale) || new Locale("pt", "BR").equals(currentLocale)) {
```

Comment:
Let's create a final static variable for the pt_BR Locale that we can use in this if statement. No need to create it every time.